File: binary_search_ll.py
'''
Searches through a linked list to find a provided key
'''
def binary_search(data,key):
    '''
    Search through the list provided and returns true if key is found
    '''
    if data.size_of() <= 1 and key != data.head.data:
        return False
    if data.head is None:
        return False

    mid = data.size_of() // 2
    current = data.at_index(mid)

    if key == current.data:
        return True
    if key < current.data:
        new_list = data.slice(0,mid)
        return binary_search(new_list, key)
    if key > current.data:
        mid += 1
        new_list = data.slice(mid)
        return binary_search(new_list, key)

File: binarySearchLL.js
const binarySearch = (list, key) => {
	if (list.sizeOf() < 1) {
		return list;
	}
	if (list.head === undefined) {
		return list;
	}

	const midPoint = Math.floor(list.sizeOf() / 2);
	const current = list.atIndex(midPoint);
	let newList;

	if (key === current.data) {
		return true;
	}
	if (key < current.data) {
		newList = list.slice(0, midPoint);
		return binarySearch(newList, key);
	}
	if (key > current.data) {
		midPoint += 1;
		newlist = list.slice(midPoint);
		return binarySearch(newList, key);
	}
};

module.exports = binarySearch;
